stan/math/rev/mat/fun/cholesky_decompose.hpp

```diff
@@ -244,9 +248,30 @@ namespace stan {
       check_symmetric("cholesky_decompose", "A", A);
 
       Eigen::Matrix<double, -1, -1> L_A(value_of_rec(A));
-      Eigen::LLT<Eigen::Ref<Eigen::MatrixXd>, Eigen::Lower> L_factor(L_A);
-      check_pos_definite("cholesky_decompose", "m", L_factor);
+      // NOTE: This should be replaced by some check that comes from a user
+
+      #ifdef STAN_GPU
+        if (L_A.rows()  > 70) {
+          L_A = L_A.selfadjointView<Eigen::Lower>();
+          viennacl::matrix<double>  vcl_L_A(L_A.rows(), L_A.cols());
+          viennacl::copy(L_A, vcl_L_A);
+          viennacl::linalg::lu_factorize(vcl_L_A);
+          viennacl::copy(vcl_L_A, L_A);
+          // TODO(Steve/Sean): Where should this check go?
+          // check_pos_definite("cholesky_decompose", "m", L_A);
```

Comment:
@seantalts the vcl here only returns the lower triangular of the cholesky. Could we modify check_pos_definite() to account for that?